### DISCUSSION TITLE
持久层是否考虑可以同时支持MySQL和Oracle？

### DISCUSSION THREAD

**Discussion by grunt1024** [apache/dolphinscheduler#6243] [Q&A]
目前来看，dao层里面的mapper文件有很多和数据库类型绑定的语句，是否考虑可以同时支持MySQL和Oracle？以做到使用者可以自由选择数据源类型。

**Comment by Narcasserun** [comment]
The persistence layer does not support MySQL and Oracle at the same time, but we can give you some solutions that support multiple data sources in the persistence layer:

1. We need all SQL statements to support SQL92 standard

2. We need a proxy service to translate sqk92 statements into corresponding data source SQL, such as MySQL and Oracle

**Comment by davidzollo** [comment]
As I know, a contributor will do this work, mybatis self support many databases, we can update some sqls to use mybatis grammar
